### ISSUE TITLE
IDE0052 should not be reported if field used with nameof

### ISSUE THREAD

**Issue by Dreamescaper** [dotnet/roslyn#32008]
**Version Used**: 
VS 2019 Preview 1.1

**Steps to Reproduce**:

NUnit example:

```csharp
        [TestCaseSource(nameof(TestCaseData))]
        public void TestCase(string expected, string actual)
        {
            Assert.That(expected, Is.EqualTo(actual));
        }

        private static readonly object[] TestCaseData =
        {
            new object[] {"expected", "actual" }
        };
```

**Expected Behavior**:

`TestCaseData` is not suggested to be removed

**Actual Behavior**:

`TestCaseData` is suggested to be removed as not used

**Comment by gafter** [comment]
If the IDE behavior is caused by a compiler diagnostic, please let us know and assign to `Area-Compiler`.

**Comment by Dreamescaper** [comment]
That's strange.
Why suggest to remove anything, if removing will lead to non-compilable code?

**Comment by mavasani** [comment]
> Why suggest to remove anything, if removing will lead to non-compilable code?

Suggestion is intended to help identify _real_ performance issues, which would be true for majority of cases this diagnostic fires. For example:

```
class C
{
  private readonly int field = ExtremelyExpensiveComputation();
  string M()
  {
    return nameof(field);
  }
}
```

**Comment by Dreamescaper** [comment]
Don't really get why would anyone use nameof of something they don't need, but anyway...

**Comment by mavasani** [comment]
Also note that the diagnostic is not just for nameof, it is meant to flag all fields/properties whose value is never _read_, for example below indicates both a performance and functional/maintenance bug:

```
class C
{
  private readonly int field = ExtremelyExpensiveComputation();
  string M(int param)
  {
    field = param;
  }
}
```

**Comment by mavasani** [comment]
FYI @Dreamescaper - https://github.com/dotnet/roslyn/pull/32675 makes the analyzer conservative by not flagging symbols which are used in nameof expressions. This fix should show in Dev16 preview3. Thanks!
